tasks/cephfs/filesystem.py

```diff
@@ -371,20 +402,34 @@ def wait_for_state(self, goal_state, reject=None, timeout=None, mds_id=None):
         Block until the MDS reaches a particular state, or a failure condition
         is met.
 
+        When there are multiple MDSs, succeed when exaclty one MDS is in the
+        goal state, or fail when any MDS is in the reject state.
+
         :param goal_state: Return once the MDS is in this state
         :param reject: Fail if the MDS enters this state before the goal state
         :param timeout: Fail if this many seconds pass before reaching goal
         :return: number of seconds waited, rounded down to integer
         """
 
-        if mds_id is None:
-            mds_id = self.get_lone_mds_id()
-
         elapsed = 0
         while True:
-            # mds_info is None if no daemon currently claims this rank
-            mds_info = self.mon_manager.get_mds_status(mds_id)
-            current_state = mds_info['state'] if mds_info else None
+
+            if mds_id is not None:
+                # mds_info is None if no daemon with this ID exists in the map
+                mds_info = self.mon_manager.get_mds_status(mds_id)
+                current_state = mds_info['state'] if mds_info else None
+                log.info("Looked up MDS state for {0}: {1}".format(mds_id, current_state))
+            else:
+                # In general, look for a single MDS
+                mds_status = self.mon_manager.get_mds_status_all()
+                states = [m['state'] for m in mds_status['info'].values()]
+                if [s for s in states if s == goal_state] == [goal_state]:
```

Comment:
Maybe my python parsing is failing me, but this looks like maybe something got copied too many times?